### ISSUE TITLE
problem with `Hecke.math_html` for `nf_elem`

### ISSUE THREAD

**Issue by ThomasBreuer** [oscar-system/Oscar.jl#737]
`Hecke.math_html` can be used to produce LaTeX format descriptions.
For the case of elements of cyclotomic fields, this looks as follows.
```
julia> F, z = cyclotomic_field(5)
(Cyclotomic field of order 5, z_5)

julia> Hecke.math_html(stdout, z+z^4)
-\zeta_{5}^{3} - \zeta_{5}^{2} - 1
```
In the context of the abelian closure of the rationals, I have to use other cyclotomic fields, in order to use the caching of these fields (which is necessary for correct comparisons of elements).
Here I do not get LaTeX format.
```
julia> FF, zz = Oscar.AbelianClosure.cyclotomic_field(abelian_closure(QQ)[1], 5)
(Cyclotomic field of order 5, ζ(5))

julia> Hecke.math_html(stdout, zz+zz^4)
-ζ(5)^{3} - ζ(5)^{2} - 1
```
Is this behaviour intended?
If yes then is there a better way to avoid it than temporarily changing `FF.S`?
```
julia> FF.S
Symbol("ζ(5)")

julia> F.S
:z_5

julia> FF.S = F.S; Hecke.math_html(stdout, zz+zz^4)
-\zeta_{5}^{3} - \zeta_{5}^{2} - 1
```

**Comment by thofma** [comment]
I think `math_html` can go (now that we have the proper printing from AA).

**Comment by fingolfin** [comment]
@thofma OK, so how would one use that to get the LaTeX Thomas is looking for?

**Comment by thofma** [comment]
Use show with the right MIME type.

**Comment by tthsqe12** [comment]
I think this works like this
```julia
julia> using AbstractAlgebra
julia> R, x = PolynomialRing(QQ, "ξ")
(Univariate Polynomial Ring in ξ over Rationals, ξ)
julia> show(stdout, (x+1)^2)
ξ^2 + 2*ξ + 1
julia> show(stdout, "text/latex", (x+1)^2)
{\xi}^{2} + 2 {\xi} + 1
```

**Comment by tthsqe12** [comment]
(I am not sure what Oscar has overridden nor do I know where the ζ comes from in Qab.) There is also
```julia
julia> AbstractAlgebra.obj_to_latex_string((x+1)^2)
"{\\xi}^{2} + 2 {\\xi} + 1"
```

**Comment by ThomasBreuer** [comment]
Thanks for the information.
(If the proper printing from AA is the recommended standard: Is it documented somewhere?)

With the proposed calls, I get
```
julia> F, z = CyclotomicField(5);

julia> show(stdout, "text/latex", z+z^4)
-\operatorname{z}_{5}^{3} - \operatorname{z}_{5}^{2} - 1
```
and
```
julia> FF, zz = Oscar.AbelianClosure.cyclotomic_field(abelian_closure(QQ)[1], 5);

julia> show(stdout, "text/latex", zz+zz^4)
-\operatorname{{\zeta}(5)}^{3} - \operatorname{{\zeta}(5)}^{2} - 1
```
As far as I understand, `\operatorname` is not a standard LaTeX command but requires the package amsopn. I have never used this package up to now.

I must admit that I found the variant
```
julia> Hecke.math_html(stdout, z+z^4)
-\zeta_{5}^{3} - \zeta_{5}^{2} - 1
```
nicer.

**Comment by thofma** [comment]
> I must admit that I found the variant
> 
> ```
> julia> Hecke.math_html(stdout, z+z^4)
> -\zeta_{5}^{3} - \zeta_{5}^{2} - 1
> ```
> 
> nicer.

We deliberately removed the random mappings between characters and latex symbols.

@tthsqe12 Do you know where those `operatorname` come from? Is this coming from the polynomial printing? We problably need to tweak the "text/latex" printing for `QabElem`.

**Comment by tthsqe12** [comment]
I am looking into this right now. Obviously AA.deunicodify and AA.underscorify could use some more work. We have to use something like `\operatorname` so that `sin(x)` doesn't look like `s*i*n(x)`. Is there a version of `\operatorname` that doesn't require a package?

**Comment by fingolfin** [comment]
For sin you could just use `\sin`, IIRC that's builtin. More generally `\mathrm{funcname}`. There is also \mathop but it is from a package. IIRC \operatorname is about spacing, though, isn't it?

**Comment by tthsqe12** [comment]
I think I will just use \mathop{\mathrm{...}} and hope for the best.
I hope no one wishes the output of the latex printer to be human-readable.

**Comment by tthsqe12** [comment]
With the linked AA pr, the issues with AA have been "solved" and printing the elements of the first `CyclotomicField(5)` does not contain the unnecessary operatorname. What is going on with the second printing is:
In nemo there is
```julia
function CyclotomicField(n::Int, s::AbstractString = "z_$n", t = "_\$"; cached = true)
```
which clearly creates the symbol "z_5". However, in oscar, there is
https://github.com/oscar-system/Oscar.jl/blob/567e4e88446ed2f942b03992712186b887ea449a/src/Rings/AbelianClosure.jl#L133
which creates the symbol "z(5)". I guess this was done so that the output can be copy and pasted back into julia?
The AA printer has limited capabilities for rendering symbols into latex. I think it is unreasonable to expect it to parse this Symbol string and print it without operatorname. @thofma ?

**Comment by tthsqe12** [comment]
@fingolfin \mathop and \operaname are about spacing, but \mathop seems to work without any package.

**Comment by thofma** [comment]
Yes, since one creates elements via `z(5)`, the last decision on this matter was to print it also as `z(5)` instead of `z_5`. Of course this is a bit odd when it comes to latex output. Since the latex output can never be parsed back easily, we might customize the "text/latex" printing to `z_5`? What do you think @tthsqe12? Or maybe someone who cares about the "text/latex" output should speak up.

**Comment by tthsqe12** [comment]
As for customizing the latex printing, I would recommend customizing the expressification, since after the expressification we are basically out of luck as the printer just sees the Symbol "z(5)". Can :latex=>true be part of the context?

**Comment by thofma** [comment]
I guess the MIME type is not part of the context? This would be the alternative.

**Comment by fingolfin** [comment]
Alternate code:
`show(stdout, MIME("text/html"), z+z^4)`

Closing this as by the comments @thofma made way up in this discussion.
